wp_api/src/reqwest_request_executor.rs

```diff
@@ -143,3 +159,227 @@ impl RequestExecutor for ReqwestRequestExecutor {
+                redirects: None,
+                reason: RequestExecutionErrorReason::HttpTimeoutError {
+                    hostname: "hostname".to_string(),
+                },
+            };
+        }
+
+        if let Some(tls_error) = error.as_tls_error() {
+            return RequestExecutionError::RequestExecutionFailed {
+                status_code: None,
+                redirects: None,
+                reason: tls_error.into(),
+            };
+        }
+
+        if let Some(io_error) = error.as_io_error() {
+            match io_error.kind() {
+                std::io::ErrorKind::ConnectionRefused => {
+                    return RequestExecutionError::RequestExecutionFailed {
+                        status_code: None,
+                        redirects: None,
+                        reason: RequestExecutionErrorReason::NonExistentSiteError {
+                            error_message: Some("Connection refused".to_string()),
+                            suggested_action: None,
+                        },
+                    };
+                }
+                std::io::ErrorKind::UnexpectedEof => {
+                    // Server terminated the connection unexpectedly
+                    return RequestExecutionError::RequestExecutionFailed {
+                        status_code: None,
+                        redirects: None,
+                        reason: RequestExecutionErrorReason::NonExistentSiteError {
+                            error_message: Some(
+                                "The server terminated the connection unexpectedly".to_string(),
+                            ),
+                            suggested_action: None,
+                        },
+                    };
+                }
+                _ => {
+                    return RequestExecutionError::RequestExecutionFailed {
+                        status_code: None,
+                        redirects: None,
+                        reason: RequestExecutionErrorReason::GenericError {
+                            error_message: error.to_string(),
+                        },
+                    };
+                }
+            }
+        }
+
+        if let Some(hyper_error) = error.as_hyper_error() {
+            return RequestExecutionError::RequestExecutionFailed {
+                status_code: None,
+                redirects: None,
+                reason: hyper_error.into(),
+            };
+        }
+
+        if let Some(dns_error) = error.as_dns_error() {
+            return RequestExecutionError::RequestExecutionFailed {
+                status_code: None,
+                redirects: None,
+                reason: dns_error.into(),
+            };
+        }
+
+        RequestExecutionError::RequestExecutionFailed {
+            status_code: None,
+            redirects: None,
+            reason: RequestExecutionErrorReason::GenericError {
+                error_message: error.to_string(),
+            },
+        }
```

Comment:
Replaced the following from original implementation since we can't have `todo!()`s in production code:
```
println!("================================================");
println!("Error: {:?}", error);
println!("Error: {:?}", error.source());
todo!("Unhandled error – aborting");
```